### PR TITLE
Work with clojure 1.10.0 ns spec

### PR DESCRIPTION
**Pull request by huahaiy**
Otherwise, will get this error:

#error {
 :cause Call to clojure.core/ns did not conform to spec.
 :data #:clojure.spec.alpha{:problems ({:path [:ns-clauses :refer-clojure :clause], :pred #{:refer-clojure}, :val :import, :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-refer-clojure :clojure.core.specs.alpha/ns-refer-clojure], :in [2 0]} {:path [:ns-clauses :require :clause], :pred #{:require}, :val :import, :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-require :clojure.core.specs.alpha/ns-require], :in [2 0]} {:path [:ns-clauses :import :classes :class], :pred clojure.core/simple-symbol?, :val [javax.xml.crypto], :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-import :clojure.core.specs.alpha/ns-import :clojure.core.specs.alpha/import-list], :in [2 1]} {:path [:ns-clauses :import :classes :package-list :classes], :reason Insufficient input, :pred clojure.core/simple-symbol?, :val (), :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-import :clojure.core.specs.alpha/ns-import :clojure.core.specs.alpha/import-list :clojure.core.specs.alpha/package-list :clojure.core.specs.alpha/package-list], :in [2 1]} {:path [:ns-clauses :use :clause], :pred #{:use}, :val :import, :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-use :clojure.core.specs.alpha/ns-use], :in [2 0]} {:path [:ns-clauses :refer :clause], :pred #{:refer}, :val :import, :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-refer :clojure.core.specs.alpha/ns-refer], :in [2 0]} {:path [:ns-clauses :load :clause], :pred #{:load}, :val :import, :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-load :clojure.core.specs.alpha/ns-load], :in [2 0]} {:path [:ns-clauses :gen-class :clause], :pred #{:gen-class}, :val :import, :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-gen-class :clojure.core.specs.alpha/ns-gen-class], :in [2 0]}), :spec #object[clojure.spec.alpha$regex_spec_impl$reify__2509 0x61892d1b clojure.spec.alpha$regex_spec_impl$reify__2509@61892d1b], :value (saml20-clj.sp (:require [clojure.data.xml :as xml] [clojure.xml :refer [parse]] [ring.util.response :refer [redirect]] [clj-time.core :as ctime] [clj-time.coerce :refer [to-timestamp]] [hiccup.core :as hiccup] [saml20-clj.shared :as shared] [saml20-clj.xml :as saml-xml] [clojure.data.zip.xml :as zf]) (:import [javax.xml.crypto] [javax.xml.crypto.dsig XMLSignature XMLSignatureFactory] [javax.xml.crypto.dom] [org.apache.xml.security Init] [org.apache.xml.security.utils Constants ElementProxy] [org.apache.xml.security.transforms Transforms] [org.apache.xml.security.c14n Canonicalizer] [javax.xml.crypto.dsig.dom DOMValidateContext] [java.security] [javax.xml.parsers DocumentBuilderFactory] [org.w3c.dom Document] [org.w3c.dom NodeList]) (:gen-class)), :args (saml20-clj.sp (:require [clojure.data.xml :as xml] [clojure.xml :refer [parse]] [ring.util.response :refer [redirect]] [clj-time.core :as ctime] [clj-time.coerce :refer [to-timestamp]] [hiccup.core :as hiccup] [saml20-clj.shared :as shared] [saml20-clj.xml :as saml-xml] [clojure.data.zip.xml :as zf]) (:import [javax.xml.crypto] [javax.xml.crypto.dsig XMLSignature XMLSignatureFactory] [javax.xml.crypto.dom] [org.apache.xml.security Init] [org.apache.xml.security.utils Constants ElementProxy] [org.apache.xml.security.transforms Transforms] [org.apache.xml.security.c14n Canonicalizer] [javax.xml.crypto.dsig.dom DOMValidateContext] [java.security] [javax.xml.parsers DocumentBuilderFactory] [org.w3c.dom Document] [org.w3c.dom NodeList]) (:gen-class))}
 :via
 [{:type clojure.lang.Compiler$CompilerException
   :message Syntax error macroexpanding clojure.core/ns at (saml20_clj/sp.clj:1:1).
   :data #:clojure.error{:phase :macro-syntax-check, :line 1, :column 1, :source saml20_clj/sp.clj, :symbol clojure.core/ns}
   :at [clojure.lang.Compiler checkSpecs Compiler.java 6971]}
  {:type clojure.lang.ExceptionInfo
   :message Call to clojure.core/ns did not conform to spec.
   :data #:clojure.spec.alpha{:problems ({:path [:ns-clauses :refer-clojure :clause], :pred #{:refer-clojure}, :val :import, :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-refer-clojure :clojure.core.specs.alpha/ns-refer-clojure], :in [2 0]} {:path [:ns-clauses :require :clause], :pred #{:require}, :val :import, :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-require :clojure.core.specs.alpha/ns-require], :in [2 0]} {:path [:ns-clauses :import :classes :class], :pred clojure.core/simple-symbol?, :val [javax.xml.crypto], :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-import :clojure.core.specs.alpha/ns-import :clojure.core.specs.alpha/import-list], :in [2 1]} {:path [:ns-clauses :import :classes :package-list :classes], :reason Insufficient input, :pred clojure.core/simple-symbol?, :val (), :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-import :clojure.core.specs.alpha/ns-import :clojure.core.specs.alpha/import-list :clojure.core.specs.alpha/package-list :clojure.core.specs.alpha/package-list], :in [2 1]} {:path [:ns-clauses :use :clause], :pred #{:use}, :val :import, :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-use :clojure.core.specs.alpha/ns-use], :in [2 0]} {:path [:ns-clauses :refer :clause], :pred #{:refer}, :val :import, :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-refer :clojure.core.specs.alpha/ns-refer], :in [2 0]} {:path [:ns-clauses :load :clause], :pred #{:load}, :val :import, :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-load :clojure.core.specs.alpha/ns-load], :in [2 0]} {:path [:ns-clauses :gen-class :clause], :pred #{:gen-class}, :val :import, :via [:clojure.core.specs.alpha/ns-form :clojure.core.specs.alpha/ns-gen-class :clojure.core.specs.alpha/ns-gen-class], :in [2 0]}), :spec #object[clojure.spec.alpha$regex_spec_impl$reify__2509 0x61892d1b clojure.spec.alpha$regex_spec_impl$reify__2509@61892d1b], :value (saml20-clj.sp (:require [clojure.data.xml :as xml] [clojure.xml :refer [parse]] [ring.util.response :refer [redirect]] [clj-time.core :as ctime] [clj-time.coerce :refer [to-timestamp]] [hiccup.core :as hiccup] [saml20-clj.shared :as shared] [saml20-clj.xml :as saml-xml] [clojure.data.zip.xml :as zf]) (:import [javax.xml.crypto] [javax.xml.crypto.dsig XMLSignature XMLSignatureFactory] [javax.xml.crypto.dom] [org.apache.xml.security Init] [org.apache.xml.security.utils Constants ElementProxy] [org.apache.xml.security.transforms Transforms] [org.apache.xml.security.c14n Canonicalizer] [javax.xml.crypto.dsig.dom DOMValidateContext] [java.security] [javax.xml.parsers DocumentBuilderFactory] [org.w3c.dom Document] [org.w3c.dom NodeList]) (:gen-class)), :args (saml20-clj.sp (:require [clojure.data.xml :as xml] [clojure.xml :refer [parse]] [ring.util.response :refer [redirect]] [clj-time.core :as ctime] [clj-time.coerce :refer [to-timestamp]] [hiccup.core :as hiccup] [saml20-clj.shared :as shared] [saml20-clj.xml :as saml-xml] [clojure.data.zip.xml :as zf]) (:import [javax.xml.crypto] [javax.xml.crypto.dsig XMLSignature XMLSignatureFactory] [javax.xml.crypto.dom] [org.apache.xml.security Init] [org.apache.xml.security.utils Constants ElementProxy] [org.apache.xml.security.transforms Transforms] [org.apache.xml.security.c14n Canonicalizer] [javax.xml.crypto.dsig.dom DOMValidateContext] [java.security] [javax.xml.parsers DocumentBuilderFactory] [org.w3c.dom Document] [org.w3c.dom NodeList]) (:gen-class))}
   :at [clojure.spec.alpha$macroexpand_check invokeStatic alpha.clj 705]}]

### CODE REVIEW

File: src/saml20_clj/sp.clj [hunk 1]
```diff
@@ -63,13 +60,15 @@
           [:ds:KeyInfo  {:xmlns:ds  "http://www.w3.org/2000/09/xmldsig#"}
            [:ds:X509Data
             [:ds:X509Certificate certificate-str]]]]
-         [:md:SingleLogoutService  {:Binding  "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST", :Location  "https://example.org/saml/SingleLogout"}]
```
Comment (ylebedeva) [inline src/saml20_clj/sp.clj:66]: It looks like the `SingleLogoutService` metadata entry has been removed here. Was it intended?

File: project.clj [hunk 2]
```diff
@@ -1,25 +1,25 @@
-(defproject kirasystems/saml20-clj "0.1.13-SNAPSHOT"
+(defproject juji/saml20-clj "0.1.13"
```
Comment (ylebedeva) [inline project.clj]: This project name change is likely coming from the fork-only diff. I think we should keep the existing `kirasystems/saml20-clj` name.

**Review by ylebedeva — CHANGES_REQUESTED** [review]
Thank you for your PR! I have left a couple of comments for your consideration.

File: project.clj (hunk 2 above)
Comment (huahaiy) [reply project.clj]: That's an oversight. Removed.

File: src/saml20_clj/sp.clj (hunk 1 above)
Comment (huahaiy) [reply src/saml20_clj/sp.clj:66]: That's change of @tvirolai, I don't care either way. Added back.

**Comment by oakes** [comment]
What's the status of this? I started looking into updating to 1.10 this morning and this was one of the deps that threw an error for me.

**Comment by dpsutton** [comment]
Is there anything I can do to help this change be merged?

**Review by ylebedeva — DISMISSED** [review]

**Comment by ylebedeva** [comment]
The build fails - other than it is good to go.
Looks like the report says we are missing `clojure.tools.reader.edn` yet I don't see references to it (a couple of options here, one being a problem with a dependency, the other that it is not a dependency problem at all but something else).
@dpsutton Running it locally to pinpoint the problem would help a lot!

**Comment by dpsutton** [comment]
just adding `[org.clojure/tools.reader "1.3.2"]` solves it for me. or at lease makes `lein test` output the same as master for me which is:

```bash
[dan@fedora saml20-clj]$ lein test
WARNING: pos-int? already refers to: #'clojure.core/pos-int? in namespace: taoensso.encore, being replaced by: #'taoensso.encore/pos-int?
WARNING: bytes? already refers to: #'clojure.core/bytes? in namespace: taoensso.encore, being replaced by: #'taoensso.encore/bytes?

lein test saml20-clj.shared-test

lein test saml20-clj.sp-test
[Fatal Error] :1:2625: JAXP00010004: The accumulated size of entities is "100,043" that exceeded the "100,000" limit set by "property".

Ran 6 tests containing 8 assertions.

```

(the shadow warnings are new with 1.10 but can be eliminated easily)

**Comment by ylebedeva** [comment]
Okay. Is this PR abandoned?
Is the only way to proceed (finish the change and bring it in) to fork the repo `huahaiy/saml20-clj` again? Cannot think of anything else here.

**Comment by MatthewDarling** [comment]
Between #8 and #12, I believe the important changes from this PR are covered. I compared the diffs and things look good.

Thanks all for contributing back!

**Pull request closed without merging**